### ISSUE TITLE
[BUG] Different SHA hashes shown for current & latest version with up-to-date image

### ISSUE THREAD

**Issue by MartinRied** [jetstack/version-checker#363]
**Describe the bug**
I am running a Bitnami OpenLDAP container [`bitnami/openldap:2.6.9`](https://hub.docker.com/layers/bitnami/openldap/2.6.9/images/sha256-e064b1d23926e9ee27e4cd1deeb4c775cff5eb1e28e6001357b83663f5e05afd).

Weirdly enough, it is still flagged as "not latest" with the difference being only in the SHA hash:
```
time="2025-04-21T08:42:09Z" level=debug msg="found: \"bitnami/openldap\"" cache=handler component=controller controller=pod module=version_getter
time="2025-04-21T08:42:09Z" level=debug msg="image is not latest bitnami/openldap: 2.6.9@sha256:d440c425dc0afa96167a323721b57df4d9a6e204b6de54ae5b4660ba4c6ab6d0 -> 2.6.9@sha256:e064b1d23926e9ee27e4cd1deeb4c775cff5eb1e28e6001357b83663f5e05afd" component=controller container=ldap controller=pod name=ldap-66dd7c4865-c5hjs namespace=ldap
Loading...
```

Based on what I saw on Docker Hub, there seems to be a confusion between the "Index Digest" (or repo digest?) `d440...` shown on the tagged image's page and the "other" (manifest? repository? ...?) digest `e064...`.
The latter doesn't seem to be just a local hash, since it's also shown on the docker hub tags listing and in the URL.

Maybe this is just a misunderstanding, then I would be grateful for a clarification.

**To Reproduce**
1. Deploy container with image 'bitnami/openldap:2.6.9'
2. Check Prometheus with the following Query 'sum by(image,current_version,latest_version) (version_checker_is_latest_version)'
3. Image is not identified as "latest"

**Expected behavior**
When using the correct latest image, I would expect it not to be flagged as "is_latest_version=No".

**Environment (please complete the following information):**
- Version-Checker: 0.9.0
- Kubernetes: 1.32.1
- Grafana: 11.6.0
- Prometheus: 3.3.0

**Comment by MartinRied** [comment]
Sorry, I probably misclassified this... should rather be a "question" than a "bug".

**Comment by davidcollom** [comment]
Hi @MartinRied, no worries about the question/bug. This is, in fact, related to the work being carried out in #355. Right now, the version-checker struggles to detect the correct version for multi-platform images (arm64/AMD64, etc.).

We are working on bringing this soon and appreciate your taking the time to raise this, as it helps us know how widespread the issue is and extra validation/testing criteria!

I'll link this issue up to our main/parent issue and add it to our V1 Milestone!
